src/unix/docker/dockerfiles/build/alpine/Dockerfile

```diff
@@ -38,4 +40,20 @@ RUN dotnet help
 
 # CLR Instrumentation Engine Build Prerequisites
 RUN apk update \
-    && apk add --no-cache libxml2-dev
\ No newline at end of file
+    && apk add --no-cache libxml2-dev
+
+# Install newer version of cmake. We need to run make commands inside the directory
+# that it is extracted to, so WORDIR is the easiest context to use.
+WORKDIR /usr/share/cmakeinst
+RUN wget -O cmake.tar.gz https://github.com/Kitware/CMake/releases/download/v3.20.2/cmake-3.20.2.tar.gz \
```

Comment:
nit: Maybe refactor the the version into an ENV command so that it can be reused here and in the `cd cmake-3.20.2` lines. Here's an example that dotnet-docker uses for this: https://github.com/dotnet/dotnet-docker/blob/720a40b1aa348ad4c9fc5575668090849ac3172b/src/monitor/5.0/alpine/amd64/Dockerfile#L8